vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.html

```diff
@@ -24,309 +27,339 @@
-    
-    selectedModuleLinkId=moduleLinkId;
-    storeX=posXEdit;
-	storeY=posYEdit;
-	$("#moduleLinkInfoLabel").text(moduleLinkName);
-	$("#moduleLinkDisplayId").val(displayLinkId);
-	$("#moduleLinkInfoLabelEdit").text(moduleLinkName);
-	$("#moduleLinkLabelEdit").val(moduleLinkName);
-	$("#moduleLinkId").val(moduleLinkId);
-	$("#moduleLinkIdValueEdit").val(moduleLinkId);
-	$("#moduleLinkRotationEdit").val(rotation);
-	$('#typeEdit option[value="'+linkType+'"]').attr("selected", "selected");
-	$('#moduleLinkIdEdit option[value="'+selectedModuleId+'"]').attr("selected", "selected");
-	resetHighlighting();     
-	$('[data-link-id="' + moduleLinkId + '"]').css("color", "#c1bb88");
-	$('div[data-link-id="' + moduleLinkId + '"]').removeClass("alert-primary");
-	$('div[data-link-id="' + moduleLinkId + '"]').addClass("alert-warning");
-	$('img[data-link-id="' + moduleLinkId + '"]').css("border", "solid 1px #c1bb88");
-	
-	$("#bgImage").on("click", function(e){
-		e.preventDefault();
-		var posX = $(this).position().left;
-		var posY = $(this).position().top;    
-		storeX = e.pageX - $(this).offset().left;
-		storeY = e.pageY - $(this).offset().top;
-		showModuleLinkEdit(editModuleLinkInfo());
-	});
-	hideLinkInfoTabs();
-    $("#editModuleLinkInfo").show();
+//link icons for module links
+var moduleLinkIconReader = new FileReader();
+var moduleLinkIcon;
+moduleLinkIconReader.onload = function(e) {
+	moduleLinkIcon = e.target.result;
+    showModuleLink(createModuleLinkInfo());
 }
 
-function makeExternalLinksEditable(linkName, linkId, externalLinkURL, posXEdit, posYEdit, displayLinkId, linkType) {
-    selectedExternalLinkId=linkId;
-    storeX=posXEdit;
-	storeY=posYEdit;
-    $("#externalLinkInfoLabel").text(linkName);
-    $("#externalLinkId").val(linkId);
-	$("#externalLinkDisplayId").val(displayLinkId);
-	$("#externalLinkInfoLabelEdit").text(linkName);
-	$("#externalLinkLabelEdit").val(linkName);
-	$("#externalLinkIdValueEdit").val(linkId);
-	$("#externalLinkURLEdit").val(externalLinkURL);
-	$('#extTypeEdit option[value="'+linkType+'"]').attr("selected", "selected");
-    resetHighlighting();
-          
-    $('[data-link-id="' + linkId + '"]').css("color", "#c1bb88");
-    $('div[data-link-id="' + linkId + '"]').removeClass("alert-primary");
-    $('div[data-link-id="' + linkId + '"]').addClass("alert-warning");
-    $('img[data-link-id="' + linkId + '"]').css("border", "solid 1px #c1bb88");
-    
-    $("#bgImage").on("click", function(e){
-		e.preventDefault();
-		var posX = $(this).position().left;
-		var posY = $(this).position().top;    
-		storeX = e.pageX - $(this).offset().left;
-		storeY = e.pageY - $(this).offset().top;
-		showExternalLinkEdit(editExternalLinkInfo());
-	});
-    hideLinkInfoTabs();
-    $("#externalLinkInfo").show();
-    $("#editExternalLinkInfo").show();
+//link icons for external links
+var externalLinkIconReader = new FileReader();
+var externalLinkIcon;

```

Comment:
see above

Reply:
It is being used outside the onload function for the links of type image.